src/MSBuild.Abstractions/MsBuildExtensions.cs

```diff
@@ -27,17 +27,17 @@ static object GetPropertyValue(object obj, string propertyName)
                       string.Format("Couldn't find property {0} in type {1}", propertyName, objType.FullName));
                 }
 
-                return propInfo.GetValue(obj, null);
+                return propInfo?.GetValue(obj, null);
             }
 
-            static PropertyInfo GetPropertyInfo(Type type, string propertyName)
+            static PropertyInfo? GetPropertyInfo(Type? type, string propertyName)
```

Comment:
can we make `type` non-null and then check that its not null on line 23 before calling `GetPropertyInfo`?

Reply:
I marked `type` as nullable because we call `type.BaseType;` on line 40 which returns nullable `Type? BaseType` and I was getting warning.
Let me rework that.